proto-gen/api_v2/codec.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021 The Jaeger Authors.
```

Comment:
maybe putting this into `/model` is better, because no other APIs can be used without model, but not vice versa.

Reply:
I also prefer model, but this is where the cracks in this approach show.  At this point what I have pushed works, but it is based on the ordering of the `init()` in two different packages.  The init method in `/proto-gen/api_v2/codec.go` has to occur after this init:

https://github.com/grpc/grpc-go/blob/master/encoding/proto/proto.go#L33

The only package in Jaeger that eventually includes `google.golang.org/grpc/encoding/proto` is `api_v2` so putting codec.go in this package guarantees that we are called second and gogoproto is the registered coded for proto with grpc.   See [this code](https://github.com/grpc/grpc-go/blob/master/encoding/encoding.go#L113) to see the registration process.

What this means is that future, unrelated changes to Jaeger could magically break this by accidentally calling the `grpc/encoding/proto` init after our init.  The good news is that CI would catch this, the bad news is that this is bad.

I don't _hate_ this change.  It allows us to stay up to date with OTel and grpc, but there are obvious drawbacks.  I support either moving forward (with additional comments in codec.go to explain the above issue) or being patient with gogoproto for a few months to see if things resolve.

Reply:
I am also fine with this, but I suggest literally pasting your explanation into the comments.

Also, if we move codec.go into model/, can we not add an _ import into it to ensure that `grpc/encoding/proto` is called first?